### PR TITLE
chore: Initialize install script to install and bootstrap nt-connect

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

**Review by kjaskiewiczz — APPROVED** [review]
I'm not a bash expert, but it looks good to me

**Review by olehermanse — CHANGES_REQUESTED** [review]
I tested it and it works! 🚀 

A couple of things to fix:

1. If you don't have jq installed the script does not handle this nicely. Ideally I'd try to get rid of the dependency. Optionally check for jq, and print a nice error message.
2. Once that happened, it left /etc/nt-connect in a bad state and the script didn't work any more;

```
ERRO[2024-02-01T13:23:21Z] Error loading configuration from file: /etc/nt-connect/nt-connect.json (Error parsing nt-connect configuration file: unexpected end of JSON input) 
ERRO[2024-02-01T13:23:21Z] Error parsing nt-connect configuration file: unexpected end of JSON input 
```

What's the desired behavior of running this script when `/etc/nt-connect` already exists?

**Comment by alfrunes** [comment]
> A couple of things to fix:
> 
> 1. If you don't have jq installed the script does not handle this nicely. Ideally I'd try to get rid of the dependency. Optionally check for jq, and print a nice error message.

`jq` is required when bootstrapping the configuration. There's also `curl` and (optionally) `systemctl` dependencies here. I'll improve the error handling for the cases where any of these are missing.

> 2. Once that happened, it left /etc/nt-connect in a bad state and the script didn't work any more;
> 
> ```
> ERRO[2024-02-01T13:23:21Z] Error loading configuration from file: /etc/nt-connect/nt-connect.json (Error parsing nt-connect configuration file: unexpected end of JSON input) 
> ERRO[2024-02-01T13:23:21Z] Error parsing nt-connect configuration file: unexpected end of JSON input 
> ```

Ouch! That's not supposed to happen.

> What's the desired behavior of running this script when `/etc/nt-connect` already exists?

The script will not attempt to overwrite any existing files except for the binary itself. It will, however, attempt to inject the tenant token into the configuration file.

File: support/install.sh [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+check_dependencies() {
+	if ! jq < /dev/null > /dev/null; then
+		echo "This script needs jq to configure your device" 1>&2
+		exit 1;
+	elif ! curl --version > /dev/null; then
+		echo "This script needs curl to setup your device" 1>&2
+		exit 1;
+	fi
+	if test -x /sbin/systemctl; then
```
Comment (olehermanse) [inline support/install.sh:99]: On my VM it's `/usr/bin/systemctl`, so this did not work. Might want to check `which systemctl` or something similar.

File: support/install.sh [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+		-H "Authorization: Bearer ${SESSION_TOKEN}" -d '@-' > /dev/null
+
+	echo "nt-connect installed successfully!"
+
+	# Enable and start systemd service
+	if test "$HAS_SYSTEMD" = "true"; then
+		systemctl enable nt-connect
+		systemctl start nt-connect
+	else
+		echo "To start the daemon, run:"
```
Comment (olehermanse) [inline support/install.sh:202]: A little bit more of a warning would have helped me here:

```suggestion
		echo "Warning: nt-connect is not running - did not find systemctl(!)"
		echo "To start the daemon, run:"
```

**Review by olehermanse — APPROVED** [review]
Tested again, looks good :)

**Pull request merged**